uportal-war/src/main/java/org/apereo/portal/portlets/lookup/PersonLookupHelperImpl.java

```diff
@@ -464,9 +489,14 @@ protected IPersonAttributes getVisiblePerson(
                         IPermission.VIEW_USER_ACTIVITY,
                         person.getName())) {
 
-            // if the user has permission, filter the person attributes according
-            // to the specified permitted attributes
-            final Map<String, List<Object>> visibleAttributes = new HashMap<String, List<Object>>();
+            // If the user has permission, filter the person attributes according
+            // to the specified permitted attributes;  the collection of permitted
+            // attributes can be different based on whether the user is trying to
+            // access information about him/herself.
+            final Set<String> permittedAttributes = person.getName().equals(principal.getKey())
```

Comment:
Yes, I was surprised about that as well.  Apparently all 3 of these are equivalent...

  - `IPerson.getUserName()`
  - `IPersonAttributes.getName()`
  - `IAuthorizationPrincipal.getKey()` (if the principal represents a user)

I added javadoc to `IAuthorizationPrincipal.getKey()` to document this fact.